### PR TITLE
Write qemu flash image.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tools/firmware.toit [hunk 1]
```diff
@@ -554,6 +555,69 @@ flash parsed/cli.Parsed -> none:
   finally:
     directory.rmdir --recursive tmp
 
+qemu_image_cmd -> cli.Command:
```
Comment (kasperl) [inline tools/firmware.toit:558]: Shouldn't this just be part of the extract command?

cli.OptionEnum "format" ["binary", "elf", "ubjson", "qemu"]

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/firmware.toit (hunk 1 above)
Comment (floitsch) [reply tools/firmware.toit]: done.

File: tools/firmware.toit [hunk 2]
```diff
@@ -389,6 +389,31 @@ extract_cmd -> cli.Command:
...
   return cli.Command "extract"
+      --long_help="""
+        Extracts the firmware image of the envelope to a file.
+
+        The following formats are supported:
+        - binary: the raw binary image. This format can be used with
+          the 'esptool' tool.
+        - elf: the ELF file of the executable. This is typically used
+          for debugging.
+        - ubjson: a UBJSON encoding of the structure/sections of the image.
```
Comment (kasperl) [inline tools/firmware.toit:400]:
```suggestion
        - ubjson: a UBJSON encoding of the sections of the image.
```

File: tools/firmware.toit [hunk 3]
```diff
@@ -389,6 +389,31 @@ extract_cmd -> cli.Command:
     cli.Flag key
         --short_help="Extract the $key part."
   return cli.Command "extract"
+      --long_help="""
+        Extracts the firmware image of the envelope to a file.
+
+        The following formats are supported:
+        - binary: the raw binary image. This format can be used with
```
Comment (kasperl) [inline tools/firmware.toit:396]:
```suggestion
        - binary: the binary app partition. This format can be used with
```

File: tools/firmware.toit [hunk 4]
```diff
@@ -389,6 +389,31 @@ extract_cmd -> cli.Command:
...
+          the 'esptool' tool.
+        - elf: the ELF file of the executable. This is typically used
+          for debugging.
+        - ubjson: a UBJSON encoding of the structure/sections of the image.
+        - qemu: a QEMU image.
```
Comment (kasperl) [inline tools/firmware.toit:401]:
```suggestion
        - qemu: a full binary image suitable for running on QEMU.
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Pull request merged**